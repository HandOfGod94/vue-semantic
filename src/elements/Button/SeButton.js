import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-button',
  props: {
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  }
})
class SeButton extends Vue {
  render (h) {
    return (
      <button class={this.classes}>
        { this.icon && <i class={`${this.icon} icon`}></i> }
        {this.label}
        {this.$slots.default}
      </button>
    )
  }

  get classes () {
    const classes = {
      [this.size]: this.size,
      compact: this.compact,
      circular: this.circular,

      ui: true,

      [`${this.float} floated`]: this.float,
      disabled: this.disabled,
      inverted: this.inverted,
      [this.color]: this.color,
      loading: this.loading,
      labeled: this.label,
      icon: this.icon,
      basic: this.basic,

      button: true
    }
    return classes
  }
}

export default SeButton
