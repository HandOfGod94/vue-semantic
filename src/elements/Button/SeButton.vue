<template>
  <component :is="as" :class="classObject">
    <i :class="`${this.icon} icon`" v-if="classObject.icon"></i>
    <slot name="label">
      {{label}}
    </slot>
    <slot></slot>
  </component>
</template>

<script>
export default {
  name: 'se-button',
  props: {
    as: { type: String, default: 'button' },
    active: Boolean,
    attach: String,
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  },
  computed: {
    classObject () {
      return {
        [this.size]: this.size,
        compact: this.compact,
        circular: this.circular,
        ui: true, // Semantic ui class
        [`${this.attach} attached`]: this.attach,
        [`${this.float} floated`]: this.float,
        disabled: this.disabled,
        inverted: this.inverted,
        [this.color]: this.color,
        loading: this.loading,
        labeled: this.label,
        icon: this.icon,
        basic: this.basic,
        button: true
      }
    }
  }
}
</script>
