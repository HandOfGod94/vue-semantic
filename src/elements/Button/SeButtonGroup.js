
export default {
  name: 'se-button-group',
  props: {
    icon: Boolean,
    width: String,
    vertical: Boolean,
    labeled: Boolean,
    color: String,
    basic: Boolean,
    size: String
  },
  computed: {
    classes: function () {
      const classes = {
        [this.size]: this.size,
        [this.width]: this.width,
        [this.color]: this.color,
        ui: true,
        vertical: this.vertical,
        labeled: this.labeled,
        icon: this.icon,
        basic: this.basic,
        buttons: true
      }
      return classes
    }
  },
  render: function (h) {
    return (
      <div class={this.classes}>
        {this.$slots.default}
      </div>
    )
  }
}
