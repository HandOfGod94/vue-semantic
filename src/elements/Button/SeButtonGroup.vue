<template>
  <div :class="classObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'se-button-group',
  props: {
    icon: Boolean,
    width: String,
    vertical: Boolean,
    labeled: Boolean,
    color: String,
    attach: String,
    basic: Boolean,
    size: String
  },
  computed: {
    classObject () {
      return {
        [this.size]: this.size,
        [this.width]: this.width,
        [this.color]: this.color,
        ui: true,
        [`${this.attach} attached`]: this.attach,
        vertical: this.vertical,
        labeled: this.labeled,
        icon: this.icon,
        basic: this.basic,
        buttons: true
      }
    }
  }
}
</script>

<style>

</style>
