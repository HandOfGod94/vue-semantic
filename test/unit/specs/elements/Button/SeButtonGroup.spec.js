import SeButtonGroup from '@/elements/Button/SeButtonGroup'
import SeButton from '@/elements/Button/SeButton'
import { mount } from '@vue/test-utils'

const labeledButtonsWrapper = {
  render (h) {
    return (
      <template>
        <SeButton label="hello"></SeButton>
        <SeButton label="world"></SeButton>
        <SeButton label="something"></SeButton>
      </template>
    )
  }
}

const buttonsWrapper = {
  render (h) {
    return (
      <template>
        <SeButton icon="play">First</SeButton>
        <SeButton icon="pause">Second</SeButton>
        <SeButton icon="mail">Third</SeButton>
      </template>
    )
  }
}

describe('SeButtonGroup snapshot test', () => {
  it('should match with exisitng snapshot', () => {
    const data = { slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeButtonGroup classes test', () => {
  it('should have "icon" class', () => {
    const data = { propsData: { icon: true } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('icon')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have number of columns in classList', () => {
    const data = { propsData: { width: 'three' }, slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('three')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "basic" class', () => {
    const data = { propsData: { basic: true }, slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('basic')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "vertical" class', () => {
    const data = { propsData: { vertical: true }, slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('vertical')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "orange" class for color', () => {
    const data = { propsData: { color: 'orange' }, slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('orange')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "huge" class for size', () => {
    const data = { propsData: { size: 'huge' }, slots: { default: buttonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('huge')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "labeled" class for size', () => {
    const data = { propsData: { labeled: true }, slots: { default: labeledButtonsWrapper } }
    const wrapper = mount(SeButtonGroup, data)
    expect(wrapper.classes()).toContain('labeled')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
