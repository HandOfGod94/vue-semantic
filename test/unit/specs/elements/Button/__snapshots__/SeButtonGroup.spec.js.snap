// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeButtonGroup classes test should have "basic" class 1`] = `
<div class="ui basic buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;

exports[`SeButtonGroup classes test should have "huge" class for size 1`] = `
<div class="huge ui buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;

exports[`SeButtonGroup classes test should have "icon" class 1`] = `<div class="ui icon buttons"></div>`;

exports[`SeButtonGroup classes test should have "labeled" class for size 1`] = `
<div class="ui labeled buttons">
    <button class="ui  labeled button">
        <!---->
        hello
    </button>
    <button class="ui  labeled button">
        <!---->
        world
    </button>
    <button class="ui  labeled button">
        <!---->
        something
    </button>
</div>
`;

exports[`SeButtonGroup classes test should have "orange" class for color 1`] = `
<div class="orange ui buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;

exports[`SeButtonGroup classes test should have "vertical" class 1`] = `
<div class="ui vertical buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;

exports[`SeButtonGroup classes test should have number of columns in classList 1`] = `
<div class="three ui buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;

exports[`SeButtonGroup snapshot test should match with exisitng snapshot 1`] = `
<div class="ui buttons">
    <button class="ui icon button">
        <i class="play icon"></i>

        First</button>
    <button class="ui icon button">
        <i class="pause icon"></i>

        Second</button>
    <button class="ui icon button">
        <i class="mail icon"></i>

        Third</button>
</div>
`;
